HexCtrl/src/Dialogs/CHexDlgDataInterpret.cpp

```diff
@@ -115,40 +146,99 @@ void CHexDlgDataInterpret::OnOK()
 	switch (refGridData->eGroup)
 	{
 	case EGroup::DIGITS:
-	{
-		LONGLONG llData;
-		ULONGLONG ullData;
-		if (!StrToInt64ExW(wstrValue, STIF_SUPPORT_HEX, &llData))
-			break;
+	{		
 		switch (refGridData->eName)
 		{
+		case EName::NAME_BINARY:
+		{			
+			CString sBinary;
+			const CString sPermitted = L"01";
+			for (int i = 0; i < wstrValue.GetLength(); i++)
+			{
+				if (sPermitted.FindOneOf(wstrValue.Mid(i, 1)) >= 0)
+					sBinary.Append(wstrValue.Mid(i, 1));
+			}
+			
+			if (sBinary.GetLength() == 8)
+			{				
+				long lData = wcstol(sBinary.GetString(), NULL, 2);
+				fSuccess = SetDigitData<UCHAR>(lData);
+			}
+		}
+		break;
+		case EName::NAME_GUID:
+		{
+			GUID guid {};
+			if (StringToGuid(wstrValue.GetString(), &guid))
+			{
+				m_pHexCtrl->SetData(m_ullOffset, guid);
+				fSuccess = true;
+			}
+		}
+		break;
 		case EName::NAME_CHAR:
+		{
+			LONGLONG llData{};
+			if (!StrToInt64ExW(wstrValue, STIF_SUPPORT_HEX, &llData))
+				break;
 			fSuccess = SetDigitData<CHAR>(llData);
-			break;
+		}
+		break;

```

Comment:
Why would you do that?
I mean remove one call to the `StrToInt64ExW(wstrValue, STIF_SUPPORT_HEX, &llData)` at the beginning of the scope (`case EGroup::DIGITS:`) and duplicate it enormous amount of time?
What's the point?

Reply:
I did this because the initial call to StrToInt64ExW() is not compatible with the GUID data type because it is not strictly a "number". The alternative was an even more difficult to follow nested switch statement. I reasoned readablity and scope isolation were more important than duplication. However, I will change it now as you wish.

Reply:
> I did this because the initial call to StrToInt64ExW() is not compatible with the GUID data type because it is not strictly a "number".

Well, then why it would be added in the "number" (`EGroup::DIGITS`) group in the first place? :)
Just create new group for that purpose, let's call it `EGroup::GUIDS` or `EGroup::MISC`.

Reply:
Please see latest checkin. Arguably a similar problem exists with NAME_LONGLONG and NAME_ULONGLONG which use a different helper method to evaluate the value. I did think of adding a new group but was trying to make the smallest number of changes to the original code.

Reply:
> I did think of adding a new group but was trying to make the smallest number of changes to the original code.

This is what groups exactly are for.
If we will put everything in one pile then what was the purpose of the groups then?